src/fides/api/api/v1/endpoints/messaging_endpoints.py

```diff
@@ -565,4 +588,190 @@ def update_basic_messaging_templates(
                 )
             )
 
-    return BulkPutMessagingTemplateResponse(succeeded=succeeded, failed=failed)
+    return BulkPutBasicMessagingTemplateResponse(succeeded=succeeded, failed=failed)
+
+
+@router.get(
+    MESSAGING_TEMPLATES_SUMMARY,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+    response_model=Page[MessagingTemplateWithPropertiesSummary],
+)
+def get_property_specific_messaging_templates_summary(
+    *, db: Session = Depends(deps.get_db), params: Params = Depends()
+) -> AbstractPage[MessagingTemplate]:
+    """
+    Returns all messaging templates, automatically saving any missing message template types to the db.
+    """
+    # First save any missing template types to db
+    save_defaults_for_all_messaging_template_types(db)
+    ordered_templates = MessagingTemplate.query(db=db).order_by(
+        MessagingTemplate.created_at.desc()
+    )
+    # Now return all templates
+    return paginate(
+        ordered_templates,
+        params=params,
+    )
+
+
+@router.get(
+    MESSAGING_TEMPLATE_DEFAULT_BY_TEMPLATE_TYPE,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+    response_model=MessagingTemplateDefault,
+)
+def get_default_messaging_template(
+    template_type: MessagingActionType,
+) -> MessagingTemplateDefault:
+    """
+    Retrieves default messaging template by template type.
+    """
+    logger.info(
+        "Finding default messaging template of template type '{}'", template_type
+    )
+    return get_default_template_by_type(template_type)
+
+
+@router.post(
+    MESSAGING_TEMPLATES_BY_TEMPLATE_TYPE,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+    response_model=Optional[MessagingTemplateWithPropertiesDetail],
+)
+def create_property_specific_messaging_template(
+    template_type: MessagingActionType,
```

Comment:
Same comment as above

Reply:
Here we already will raise a 400 on `MessagingTemplateValidationException` so we're covered when an invalid template_type is provided 👍